src/components/Navigation/SideBar/MenuItem.tsx

```diff
@@ -92,63 +27,87 @@ const MenuItem = forwardRef<HTMLAnchorElement, MenuItemProps>(
   ({ currentUrl, icon, name, link, onClick, disabled = false }, ref) => {
     const isCurrentUrl = currentUrl?.includes(link) ?? false;
     const { isOpen } = useSideBar();
+    const [iconColor, setIconColor] = useState(
+      disabled
+        ? colors.interactive.disabled__text.rgba
+        : colors.interactive.primary__resting.rgba
+    );
+
+    const canNavigate = useMemo(
+      () => !disabled && !isCurrentUrl,
+      [disabled, isCurrentUrl]
+    );
 
-    const iconColor = useMemo(() => {
-      if (disabled) {
-        return colors.interactive.disabled__text.rgba;
-      } else if (currentUrl) {
-        return colors.interactive.primary__resting.rgba;
-      } else {
-        return colors.interactive.primary__resting.rgba;
+    const handleOnMouseEnter = () => {
+      if (!disabled && !isCurrentUrl) {
+        setIconColor(colors.text.static_icons__default.rgba);
       }
-    }, [currentUrl, disabled]);
+    };
+
+    const handleOnMouseLeave = () =>
+      setIconColor(
+        disabled
+          ? colors.interactive.disabled__text.rgba
+          : colors.interactive.primary__resting.rgba
+      );
 
     const handleOnClick = () => {
-      if (!isCurrentUrl && !disabled) {
+      if (canNavigate) {
         onClick();
       }
     };
 
     if (isOpen) {
       return (
-        <Container
+        <Link
```

Comment:
You should be able to do the styling of these items (icon colors specifically) in the `const Link = styled...` code instead I think. That way we avoid rerenders when users hover

Reply:
Yeah, this solution is not great. Icon color is not a style, but a prop. I could override it by setting svg fill attribute in the styling. Is that better?

Reply:
How did you try setting the fill of the svg? Should be possible (IIRC if you define color in the <Icon ../> then it will always use that one so make sure you don't set an initial color)

Reply:
Moving the icon color styles broke my tests